src/NosnocSolver.m

```diff
@@ -137,11 +137,10 @@ function set(obj, type, val)
             if strcmp(obj.solver_options.solver, 'ipopt')
                 last_stats = stats.solver_stats(end);
                 converged = 0;
-                if isfield(last_stats, 'iterations')
+                if isfield(last_stats, 'iterations') && ~isempty(last_stats.iterations)
                     inf_pr = last_stats.iterations.inf_pr(end);
                     inf_du = last_stats.iterations.inf_du(end);
-                    if inf_pr < obj.solver_options.opts_casadi_nlp.ipopt.tol && inf_du < obj.solver_options.opts_casadi_nlp.ipopt.tol ...
-                            && stats.complementarity_stats(end) < 10 * obj.solver_options.comp_tol
+                    if stats.complementarity_stats(end) < 10 * obj.solver_options.comp_tol
```

Comment:
why the *10?

Reply:
That is a good question. I think this was added at some point to give a buffer because of numerics but `*10` is probably too large.